eng/testing/tests.mobile.targets

```diff
@@ -24,7 +24,7 @@
     <DotnetPgoToolPath>$([MSBuild]::NormalizePath('$(DotnetPgoToolDir)', 'dotnet-pgo'))</DotnetPgoToolPath>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(EnableAggressiveTrimming)' == 'true' or '$(EnableSoftTrimming)' == 'true'">
+  <PropertyGroup Condition="('$(EnableAggressiveTrimming)' == 'true' or '$(EnableSoftTrimming)' == 'true') and '$(BuildTestsOnHelix)' != 'true'">
```

Comment:
Because this is in the context of the build machine and not the one on helix. Said another way, we want to support trimming on the build machine for configurations that can handle it.

Reply:
I think it would be good to have that as a comment above the property group, to provide more context